pkg/storage-init/storage-init.sh

```diff
@@ -108,14 +108,25 @@ if P3=$(findfs PARTLABEL=P3) && [ -n "$P3" ]; then
        P3_FS_TYPE=zfs
     fi
 
-    #For systems with ext3 filesystem, try not to change to ext4, since it will brick
-    #the device when falling back to old images expecting P3 to be ext3. Migrate to ext4
-    #when we do usb install, this way the transition is more controlled.
-    #Any fsck error (ext3 or ext4), will lead to formatting P3 with ext4
-    if { [ "$P3_FS_TYPE" != ext3 ] && [ "$P3_FS_TYPE" != ext4 ]; } || ! "fsck.$P3_FS_TYPE" -y "$P3" ; then
-       echo "P3 partition $P3 of type $P3_FS_TYPE appears to be corrupted, recreating it as $P3_FS_TYPE_DEFAULT"
-       INIT_FS=1
-       P3_FS_TYPE="$P3_FS_TYPE_DEFAULT"
+    if [ "$P3_FS_TYPE" = zfs_member ]; then
+        if ! chroot /hostfs zpool import -d "$P3" persist; then
+            echo "$(date -Ins -u) Cannot import persist pool on P3 partition $P3 of type $P3_FS_TYPE, recreating it as $P3_FS_TYPE_DEFAULT"
```

Comment:
Do you hit this case during testing?

If we think we'll hit it in production it should be a an error log (failed import) plus a warning log about recreating /persist.  But AFAIK we don't have a good way to have shell scripts cause structuted log output.

Reply:
No, I did not hit that. The reason to have it: we want to check here, do we need to recreate FS. So, firstly we check what FS we have, secondly we check if exist one hit our assumption. We have few words in output to track the actions